docs/source/java/cdata.rst

```diff
@@ -220,4 +221,252 @@ Let's create a Java class to test our bridge:
+
+    import java.util.Arrays;
+
+    public class ToBeCalledByCpp {
+        final static BufferAllocator allocator = new RootAllocator();
+
+        public static void fillVector(long schemaAddress, long arrayAddress){
+            try (ArrowArray arrow_array = ArrowArray.wrap(arrayAddress);
+                 ArrowSchema arrow_schema = ArrowSchema.wrap(schemaAddress) ) {
+                Data.exportVector(allocator, populateFieldVectorToExport(), null, arrow_array, arrow_schema);
+            }
+        }
+
+        public static FieldVector populateFieldVectorToExport(){
+            IntVector intVector = new IntVector("int-to-export", allocator);
+            intVector.allocateNew(3);
+            intVector.setSafe(0, 1);
+            intVector.setSafe(1, 2);
+            intVector.setSafe(2, 3);
+            intVector.setValueCount(3);
+            System.out.println("[Java] FieldVector: \n" + intVector);
+            return intVector;
+        }
+
+        public static void fillVectorSchemaRoot(long schemaAddress, long arrayAddress){
+            try (ArrowArray arrow_array = ArrowArray.wrap(arrayAddress);
+                 ArrowSchema arrow_schema = ArrowSchema.wrap(schemaAddress) ) {
+                Data.exportVectorSchemaRoot(allocator, populateVectorSchemaRootToExport(), null, arrow_array, arrow_schema);
+            }
+        }
+
+        public static VectorSchemaRoot populateVectorSchemaRootToExport(){
+            IntVector intVector = new IntVector("age-to-export", allocator);
+            intVector.setSafe(0, 10);
+            intVector.setSafe(1, 20);
+            intVector.setSafe(2, 30);
+            VectorSchemaRoot root = new VectorSchemaRoot(Arrays.asList(intVector));
+            root.setRowCount(3);
+            System.out.println("[Java] VectorSchemaRoot: \n" + root.contentToTSVString());
+            return root;
+        }
+    }
+
+
+Build the JAR and copy it to the C++ project.
+
+.. code-block:: shell
+
+    $ mvn clean install
+    $ cp target/cpptojava-1.0-SNAPSHOT-jar-with-dependencies.jar <c++ project path>/cpptojava.jar
+
+**C++ Side**
+
+This application uses JNI to call Java code, but transfers data (zero-copy) via the C Data Interface instead.
+
+.. code-block:: cpp
+
+    #include <iostream>
+    #include <arrow/api.h>
+    #include <arrow/c/bridge.h>
+    #include <jni.h>
+
+    JNIEnv *CreateVM(JavaVM **jvm) {
+        JNIEnv *env;
+        JavaVMInitArgs vm_args;
+        JavaVMOption options[2];
+        options[0].optionString = "-Djava.class.path=cpptojava.jar"; // java jar name
+        options[1].optionString = "-DXcheck:jni:pedantic";
+        vm_args.version = JNI_VERSION_1_8;
+        vm_args.nOptions = 2;
+        vm_args.options = options;
+        int status = JNI_CreateJavaVM(jvm, (void **) &env, &vm_args);
+        if (status < 0) {
+            std::cout << "\n<<<<< Unable to Launch JVM >>>>>\n" << std::endl;
+            std::abort();
```

Comment:
Requires `#include <cstdlib>`

Reply:
Actually I suppose the `abort` isn't necessary, maybe just explicitly return `nullptr` and it'll exit in main()

Reply:
Changed